sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -753,6 +753,14 @@ object SQLConf {
       .checkValue(_ > 0, "The initial number of partitions must be positive.")
       .createOptional
 
+  val COLLATION_ENABLED =
```

Comment:
we should make it lazy val, in case the default value (Utils.isTesting) gets materialized before the test environment is set up.